packages/core-data/src/actions.js

```diff
@@ -789,6 +789,11 @@ export const __experimentalSaveSpecifiedEntityEdits =
 				editsToSave[ edit ] = edits[ edit ];
 			}
 		}
+
+		// Provide recordKey to saveEntityRecord to avoid creating a new record
+		// and instead update the existing record.
+		editsToSave.id = recordId;
```

Comment:
Another option is we revert this change and instead pass the `id` in the call to `saveSpecifiedEntityEdits` from the Navigation sidebar hooks.

I noticed that `__experimentalSaveSpecifiedEntityEdits` is used elsewhere and I'm nervous about modifying it this close to a release.